### ISSUE TITLE
Trailing newline is stripped from template when rendering.

### ISSUE THREAD

**Issue by memotype** [pallets/jinja#848]
## Expected Behavior
When rendering a file that ends in a newline (\n) character, I would expect that the rendered output would also end in a newline character. It is a standard convention on UNIX-like systems to end text files with a trailing newline.

## Actual Behavior
The final newline is stripped from the rendered output.

## Example
```
>>> c
u'foo is {{ foo }}\nquux is {{ quux }}\nasdf\n'
>>> t = jinja2.Template(c)
>>> t.render(foo="bar", quux="baz")
u'foo is bar\nquux is baz\nasdf'
>>> s = t.stream()
>>> s.next()
u'foo is \nquux is \nasdf'
>>> s.next()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ifreeman/.local/lib/python2.7/site-packages/jinja2/environment.py", line 1271, in __next__
    return self._next()
StopIteration
>>>
```

## Your Environment
* Python version: 2.7.14+ (Debian buster/testing)
* Jinja version: 2.10 (via pip)

**Comment by davidism** [comment]
Pass `keep_trailing_newline=True` to the `Environment`: http://jinja.pocoo.org/docs/api/#jinja2.Environment

**Comment by memotype** [comment]
Ok, thanks for the response, I wasn't aware that was an option, but it seems like this should be the default. It seems odd to assume the user wants to modify the input in unexpected ways that have nothing to do with the template language. In fact, the reason I'm even opening this issue is because I'm using shinto-cli (https://github.com/istrategylabs/shinto-cli) to parameterize some files in a build job, and I noticed that it was outputting files without newlines, so when I append to them I have lines swished together on one line. At first I was going to open an issue with shinto-cli until I glanced through the source, and then tried a basic jinja2 template from the python REPL myself and realized this is just default jinja2 behavior.

But, if the jinja2 devs prefer to surprise their users by modifying the input in unexpected, unintuitive and non-standard ways, I can open an issue with shinto-cli to ask that they add the keep_trailing_newline=True option to all of their Environment calls...
